.github/workflows/mutation-testing-annotations.yaml

```
@@ -0,0 +1,57 @@
+# yamllint disable rule:line-length
+# yamllint disable rule:braces
+
+name: Infection
+
+on:
+    pull_request:
+    push:
+        branches:
+            - main
+            - master
+
+jobs:
+    tests:
+        runs-on: ubuntu-latest
+
+        strategy:
+            matrix:
+                php-version: ['7.4']
+
+        name: Mutation Testing Code Review Annotations ${{ matrix.php-version }}
+
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v2
+
+            - name: Setup PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: ${{ matrix.php-version }}
+                  tools: composer:v2
+
+            - name: Get composer cache directory
+              id: composer-cache
+              run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            - name: Cache dependencies
+              uses: actions/cache@v2
+              with:
+                  path: ${{ steps.composer-cache.outputs.dir }}
+                  key: composer-${{ runner.os }}-${{ matrix.php-version }}-${{ hashFiles('**/composer.*') }}-${{ matrix.dependencies }}
+                  restore-keys: |
+                      composer-${{ runner.os }}-${{ matrix.php-version }}-${{ hashFiles('**/composer.*') }}-
+                      composer-${{ runner.os }}-${{ matrix.php-version }}-
+                      composer-${{ runner.os }}-
+                      composer-
+
+            - name: Install dependencies
+              run: |
+                  composer update --no-interaction --prefer-dist --no-progress
+
+            - name: Run Infection for added files only
+              run: |
+                  source .github/github-functions.sh;
+                  INFECTION_FLAGS=$(get_infection_pr_flags);
+                  echo "INFECTION FLAGS: ${INFECTION_FLAGS}";
+                  php bin/infection -j2 $INFECTION_FLAGS;
```

Assuming this is a PR-only workflow:
```suggestion
                  php bin/infection -j2 --filter=$(git diff origin/"${GITHUB_BASE_REF:-master}" --diff-filter=A --name-only | grep src/ | paste -sd "," -) --ignore-msi-with-no-mutations --only-covered 
```

Alternatively, this can be put in a separate step, and then this step could depend on `if:` previous step had anything.